Rube-Goldberg Class Engineering Project (Gone OSHA)

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![rube](https://user-images.githubusercontent.com/55075747/169677140-fafe6462-0a6a-4a62-8356-da05e87b3680.png)

Welcome to the Rube-Goldberg Class PowerPlant! The only testing ship designed by a fuckup engineering grad student who built it with their bare hands! Learn to set up a TEG with the RUBE! Holodisks explaining a specialized and easy to use/learn atmos system! Hobbled together spaces cramped in size! Enjoy your hell of a time on this floating OSHA violation with your buddy the Chief at Engineering!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Currently there are no ships that are designed completely around giving engineers toys to play with. This ship comes with a garage full of equipment to make most of the content available to them. This ship is easy to learn, especially if you're unfamiliar with atmos or the TEG. This comes with a set of 6 Holodisks on the operation and special mechanics of this ship, and a special 7th one hidden somewhere! Wow your players with the new RUBE. And give me cookies because it's a good addition to our roster.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added the Rube-Goldberg Class Engineering Project ship.
imageadd: Added a steering wheel (sprite by Dexxiol)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Adds the base layout

Titular

Latentish committed:
Adds some room detail

-adds detail to some rooms

Latentish committed:
Merge branch 'shiptest-ss13:master' into EngiTribe---Tide-Class-Construction-Project

Latentish committed:
Small changes

Latentish committed:
Installs the atmos piping.

-Remodels the atmos room a bit
-adds 2 more gas chambers.

Latentish commented:
Might expand the engine room so that there's more room for construction projects.

Latentish commented:
While I don't intend to make it in this PR, I would really enjoy a way to transfer power from one ship to another. Even if it was just SMES that can be moved around while still holding power. Preferably it'd be more like batteries you fill up with power before hooking it up to a SMES. Or even a wireless power transfer device you use by touching a relay to another ship's power grid. Or a power relay system that works when you aren't docked with the power providing ship, who sells it to you for research points, credits, cargo points, vendor points, or custom store. (I know that last one is out there, sue me), but any of these would give this ship something more to do with all that power they'll be making.
Without this existing my current idea is to give it some circuit kits for higher power consumption products. Maybe something to promote planet colonization type construction projects. Because as is, it's kind of a power plant for the sake of Engineering and Atmos Tech gameplay. Which I think is fine on its own, but for players like me who've done that quite a lot already, I'd like more goals. And I think that colony building goals would suit this ship well.

Latentish committed:
Adds a Steering wheel sprite

Placeholder sprite until I sucker someone else into making it. The Chief Engineer really likes cars.

MusselMain commented:
Gud damn this thing looks like what happens when engineers don't follow the instructions, I love it

MusselMain commented:
> While I don't intend to make it in this PR, I would really enjoy a way to transfer power from one ship to another. Even if it was just SMES that can be moved around while still holding power. Preferably it'd be more like batteries you fill up with power before hooking it up to a SMES. Or even a wireless power transfer device you use by touching a relay to another ship's power grid. Or a power relay system that works when you aren't docked with the power providing ship, who sells it to you for research points, credits, cargo points, vendor points, or custom store. (I know that last one is out there, sue me), but any of these would give this ship something more to do with all that power they'll be making. Without this existing my current idea is to give it some circuit kits for higher power consumption products. Maybe something to promote planet colonization type construction projects. Because as is, it's kind of a power plant for the sake of Engineering and Atmos Tech gameplay. Which I think is fine on its own, but for players like me who've done that quite a lot already, I'd like more goals. And I think that colony building goals would suit this ship well.

Isn't the Bubble designed to be a colony ship? I mostly agree, sometimes new ships should help create different playstyles, however I also feel that your argument can be used against ships like the pill series, which technically all follow the same main idea of hardcore mode, but for those I think everyone gets a bit of fun out of them, so it doesn't matter as much, personally I feel that not every addition should be ground shatteringly unique, but I do get where you're coming from

Latentish commented:
> Isn't the Bubble designed to be a colony ship? I mostly agree, sometimes new ships should help create different playstyles, however I also feel that your argument can be used against ships like the pill series, which technically all follow the same main idea of hardcore mode, but for those I think everyone gets a bit of fun out of them, so it doesn't matter as much, personally I feel that not every addition should be ground shatteringly unique, but I do get where you're coming from

Are you Referring to the idea of adding colony elements, or adding a way to mass store and move electricity? When it comes to the colony element, the idea was that an engineer crew could go to a planet, build a shack, and use the ship as a power factory. But reinforcing that idea ends at giving them a note or wood.

Latentish committed:
Most of the work.

-Someone for the love of god fix transit tubes. Mark please BB.
-Codes the shuttle into a shiny new Json.
-Adds wiring, atmos, and a lot of supplies.
-Remodels certain areas.

Latentish commented:
I'll fix those stray conveyor belts later

Apogee-dev commented:
what's the lore for this monstrosity anyway? it really smacks of a particularly anarchistic GEC ship to me, and you might want to theme the outfits and job titles to match. The only ship that uses those and their uniforms right now is the kugelblitz anyway.

Latentish commented:
> 

The idea I had originally was making a ship that was comprised of a bunch of small cramped compartments welded together. I eventually blocked out shapes that worked for what I wanted and it took on this sort of "Engineering Graduate gets a grant to make their own ship by hand" sort of design. They blew most of the money on equipment, and just shoved it in storage or wherever they could fit it. And they spent the budget to cover everything they could need, at the cost of a single solid hull. The job names are similarly themed as if they were named by this same student. "Chief at Engineering", "Chief at Boozineering", "Engineering/Atmospherics/Mining Specialist". Names that get the point across aren't the common vernacular, but aren't very inspired.
It's supposed to feel like a huge ship you're on, but with only cramped spaces to move around in.

Latentish committed:
Detailing and Remodelling

-Remodels the bar backroom
-Replaces all walls
-Hides the drip
-Tunes some of the equipment
-Lot of detailing

Latentish committed:
Last Touchups and Holodisks

-Adds a series of holodisks to explain and help with operation.
-Adds some finishing touches.

Latentish commented:
![image](https://user-images.githubusercontent.com/55075747/169192048-208dc3bc-bc20-4c6f-b61b-0e076e63e659.png)
![image](https://user-images.githubusercontent.com/55075747/169192067-72dbe710-bf02-440d-9739-7d13eed6882d.png)
Hm

Latentish committed:
Stray wall

-Changes a stray wall.

Latentish committed:
Remodels the Dorms

-Remodels the dorms =)

Latentish committed:
Adds Decals and a Pet

-Adds a pet to the bar.
-Adds decals and detailing to the floors.

Latentish committed:
I hate underdoor tiles

By god's light I smite you.
Fuck you underdoor walls.

Latentish committed:
Minor text changes

-Corrects two errors in the holodisks.

Latentish commented:
First day of TM got a lot of good feedback. Nothing in the ship is insanely broken or unreasonable. So it should be good to stay TM'd to get more feedback. Do not merge this until I get a patch up.

Latentish committed:
TM Patch 1/--

- Replaces the sprite for the steering wheel with a new sprite from Dexxiol.
-Replaces the bar sign.
-Replaces the conveyors with disposals.
-Adds 3 of each basic level stock part.
-Adds a second shuttle expansion permit for building a second shuttle.
-Tweaks the names of the holotapes.
-Replaces holobarrier and tinyfan combos with an atmos holobarrier and signage.
-Makes the doorways of the atmos tanks breatheable.

Latentish commented:
Last patch also added the missing Particle Accelerator part, and fixed the suit storage unit that wasn't accepting items.

Latentish committed:
Minor fix

- Extends holodisk delays
- Fixes holodisk names

Latentish committed:
Fixes Bar job slots

-There can be only one.

Latentish committed:
Small fix

-Really should wait on doing this until I get a good number of the. But fuck you this github autosquashes.
-Swaps out a plating for a reinforced plating.

Latentish committed:
Wrong machine frame

-Fixed the machine frame.

Latentish committed:
-TM Patch 2/--

-Adds some extra circuits for extra thrusters and SMESs.
-Adds a secret.

Latentish committed:
Adds server

-Adds an rnd server for the ship.

Vexylius commented:
this ship is scary
don't feed it to your children

#INFkW/delamination

Latentish commented:
> this ship is scary don't feed it to your children
> 
> #INFkW/delamination

Almost a haiku

Dexxiol commented:
what the actual fuck is this ship jesus christ

Apogee-dev commented on _maps/configs/engineeringmess.json:
```diff
@@ -0,0 +1,37 @@
+	"job_slots": {
+		"Chief at Engineering": {
+			"outfit": "/datum/outfit/job/ce",
+			"slots": 1
```
"outfit": "/datum/outfit/job/ce",
"officer": true,
"slots": 1

there's a flag for officer positions. the first position serves as captain but adding an officer flag gives him the officer symbol on the manifest.

Latentish replied:
Good catch.

Latentish committed:
Makes the CE an officer

-Titular
